backend/src/test/java/org/uresti/pozarreal/service/impl/HousesServiceImplTests.java

```diff
@@ -0,0 +1,232 @@
+import org.uresti.pozarreal.model.Street;
+import org.uresti.pozarreal.repository.HousesRepository;
+import org.uresti.pozarreal.repository.RepresentativeRepository;
+import org.uresti.pozarreal.repository.StreetRepository;
+
+import java.security.Principal;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.UUID;
+
+public class HousesServiceImplTests {
+
+    @Test
+    public void givenAnNullStringId_whenGetHousesByStreet_thenReturnEmptyList() {
+        // Given:
+        HousesRepository housesRepository = Mockito.mock(HousesRepository.class);
+        StreetRepository streetRepository = Mockito.mock(StreetRepository.class);
+        RepresentativeRepository representativeRepository = Mockito.mock(RepresentativeRepository.class);
+        SessionHelper mockSessionHelper = Mockito.mock(SessionHelper.class);
+        Principal principal = Mockito.mock(Principal.class);
+
+        HousesServiceImpl housesService = new HousesServiceImpl(
+                housesRepository,
+                streetRepository,
+                representativeRepository,
+                mockSessionHelper);
+
+        List<org.uresti.pozarreal.model.House> houses = new LinkedList<>();
+
+        // When:
+        Mockito.when(housesRepository.findAllByStreetOrderByNumber("123")).thenReturn(houses);
+
+        // Then:
+        List<House> housesByStreet = housesService.getHousesByStreet("123", mockSessionHelper.getLoggedUser(principal));
+        Assertions.assertThat(housesByStreet).isEmpty();
+    }
+
+    @Test
+    public void givenAnHousesWithOneElement_whenGetHousesByStreet_thenReturnListWithOneElement() {
+        // Given:
+        HousesRepository housesRepository = Mockito.mock(HousesRepository.class);
+        StreetRepository streetRepository = Mockito.mock(StreetRepository.class);
+        RepresentativeRepository representativeRepository = Mockito.mock(RepresentativeRepository.class);
+        SessionHelper mockSessionHelper = Mockito.mock(SessionHelper.class);
+        Principal principal = Mockito.mock(Principal.class);
+
+        HousesServiceImpl housesService = new HousesServiceImpl(
+                housesRepository,
+                streetRepository,
+                representativeRepository,
+                mockSessionHelper);
+
+        List<org.uresti.pozarreal.model.House> houses = new LinkedList<>();
+
+        org.uresti.pozarreal.model.House house = org.uresti.pozarreal.model.House.builder()
+                .id("1")
+                .notes("hello")
+                .street("Street 1")
+                .number("123456")
+                .chipsEnabled(true)
+                .build();
+
+        houses.add(house);
+
+        Mockito.when(housesRepository.findAllByStreetOrderByNumber("123")).thenReturn(houses);
+
+        // When:
+        List<House> housesByStreet = housesService.getHousesByStreet("123", mockSessionHelper.getLoggedUser(principal));
```

Comment:
don't use the mock objects during testing .. the mock instances should be used only inside the methods you're testing.

I mean, instead of using  
```
 mockSessionHelper.getLoggedUser(principal)
```
you should be passing an instance of a LoggedUser